src/lib.rs

```diff
@@ -41,19 +44,32 @@ mod observer;
 pub fn run(config: &Config) -> Result<(), Box<Error>> {
     let mut event_loop =
         tokio_core::reactor::Core::new().expect("Could not initialize tokio event loop");
-    let origin = Ethereum::new(
+    let mut origin = Ethereum::new(
         config.origin_endpoint(),
         config.origin_validator_address(),
         config.origin_polling_interval(),
         Box::new(event_loop.handle()),
     );
-    let auxiliary = Ethereum::new(
+    let mut auxiliary = Ethereum::new(
         config.auxiliary_endpoint(),
         config.auxiliary_validator_address(),
         config.auxiliary_polling_interval(),
         Box::new(event_loop.handle()),
     );
 
+    let origin_block_reporter = Reactor::BlockReporter {
+        block_store_address: config.origin_block_store_address(),
+        validator_address: config.origin_validator_address(),
+    };
+
+    let auxiliary_block_reporter = Reactor::BlockReporter {
+        block_store_address: config.auxiliary_block_store_address(),
+        validator_address: config.auxiliary_validator_address(),
+    };
+
+    origin.register_observer(origin_block_reporter);
+    auxiliary.register_observer(auxiliary_block_reporter);
```

Comment:
Wouldn't the observer be a better place to register the reactors? The `lib::run()` function should do as little as possible to get the program started. And the reactor registration can potentially grow quite large.

Reply:
Observer will have to a lot of config in the parameter then, let's discuss which is the correct place to do this.

Reply:
In my PR I already pass the config object to observer's `run` method. Maybe we should have a quick call to discuss the layout.